### PR TITLE
Switching to run JupyterLite Notebooks

### PR DESCRIPTION
**Pull request by ns-rse**
* Adds some package dependencies (`ipywidgets` and `ipysheet`).
* Adds `pre-commit.ci`
* Updates documentation
  * Use of pre-commit in contributing.
  * PyPI releases
  * Development version release to PyPI
  * Switch usage page to describe use of pgfinder-jupyterlite & include figures
* Removes logging and only does so for `find_pg`

### CODE REVIEW

**Comment by TheLostLambda** [comment]
Hi @ns-rse ! Let me know if you need a code review at some point! I assume you're keen on tackling bits of the caching issue still? Maybe with some of those JS hacks? But let me know if there is anything I can do to help!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -36,3 +36,10 @@ repos:
     hooks:
       - id: ruff
         args: [--fix, --exit-non-zero-on-fix]
+
+ci:
+  autofix_prs: true
+  autofix_commit_msg: '[pre-commit.ci] Fixing issues with pre-commit'
+  autoupdate_schedule: weekly
+  autoupdate_commit_msg: '[pre-commit.ci] pre-commit automatically updated.'
+  # skip: [] # Optionally list ids of hooks to skip on CI
```
Comment (bobturneruk) [inline .pre-commit-config.yaml:45]: Is there a separate workflow file for this, or does GHA pick it up from here?

File: data/masses/c_diff_monomer_masses.csv [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿Structure,Monoisotopicmass
+Structure,Monoisotopicmass
```
Comment (bobturneruk) [inline data/masses/c_diff_monomer_masses.csv:1]: What's changed here?

File: data/masses/e_coli_monomer_masses.csv [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿Structure,Monoisotopicmass
+Structure,Monoisotopicmass
```
Comment (bobturneruk) [inline data/masses/e_coli_monomer_masses.csv:1]: What's changed here?

File: pyproject.toml [hunk 4]
```diff
@@ -23,6 +23,8 @@ keywords = [
 ]
 requires-python = ">=3.8"
 dependencies = [
+  "ipywidgets",
+  "ipysheet",
```
Comment (bobturneruk) [inline pyproject.toml:27]: I guess we could seperate these out under a "notebooks" group? Maybe something for later...

File: tests/resources/data/masses/c_diff_monomer_masses.csv [hunk 5]
```diff
@@ -1,4 +1,4 @@
-﻿Structure,Monoisotopicmass
+Structure,Monoisotopicmass
```
Comment (bobturneruk) [inline tests/resources/data/masses/c_diff_monomer_masses.csv:1]: What's changed? Line ending?

File: tests/resources/data/masses/e_coli_monomer_masses.csv [hunk 6]
```diff
@@ -1,4 +1,4 @@
-﻿Structure,Monoisotopicmass
+Structure,Monoisotopicmass
```
Comment (bobturneruk) [inline tests/resources/data/masses/e_coli_monomer_masses.csv:1]: What's changed? Line ending?

**Review by bobturneruk — APPROVED** [review]
Looks good, but I somehow haven't clocked how it removes the logging?

File: data/masses/c_diff_monomer_masses.csv (hunk 2 above)
Comment (ns-rse) [reply data/masses/c_diff_monomer_masses.csv:1]: No idea, couldn't see it myself either. :shrug:

**Comment by bobturneruk** [comment]
I'm still getting an error when running the notebook. Presumably this is because it's still loading version 0.0.3a3. If I change to 0.0.3-rc1 there's no wheel so it doesn't work. What do we need to do next, please @ns-rse?

File: data/masses/e_coli_monomer_masses.csv (hunk 3 above)
Comment (ns-rse) [reply data/masses/e_coli_monomer_masses.csv:1]: Ditto :shrug:

File: pyproject.toml (hunk 4 above)
Comment (ns-rse) [reply pyproject.toml:27]: If the Notebook here is no longer going to be used I should remove these and leave them solely as dependencies of [`mesnage-org/pgfinder-jupyterlite`](https://github.com/mesnage-org/pgfinder-jupyterlite).

File: tests/resources/data/masses/c_diff_monomer_masses.csv (hunk 5 above)
Comment (ns-rse) [reply tests/resources/data/masses/c_diff_monomer_masses.csv:1]: :shrug:

File: tests/resources/data/masses/e_coli_monomer_masses.csv (hunk 6 above)
Comment (ns-rse) [reply tests/resources/data/masses/e_coli_monomer_masses.csv:1]: :shrug:

**Comment by ns-rse** [comment]
There is no `-rc1` candidate yet so no wheel would be available.

Is it the same `OSError`?

```
---------------------------------------------------------------------------
OSError                                   Traceback (most recent call last)
Cell In[2], line 12
      9 from ipysheet import from_dataframe
     10 import pandas as pd
---> 12 from pgfinder import matching, pgio, validation
     14 # Get list of modifications
     15 allowed_mods = validation.allowed_modifications()

File /lib/python3.11/site-packages/pgfinder/__init__.py:6
      4 import yaml
      5 from pkgutil import get_data
----> 6 from pgfinder.logs.logs import setup_logger, LOGGER_NAME
      7 from pgfinder.utils import dict_to_decimal
      9 LOGGER = setup_logger()

File /lib/python3.11/site-packages/pgfinder/logs/logs.py:12
      9 # pylint: disable=assignment-from-no-return
     11 start = datetime.now()
---> 12 LOG_CONFIG = logging.basicConfig(
     13     filename=Path().cwd().stem +
f"-{start.strftime('%Y-%m-%d-%H-%M-%S')}.log", filemode="w"
     14 )
     15 LOG_FORMATTER = logging.Formatter(
     16     fmt="[%(asctime)s] [%(levelname)-8s] [%(name)s]
%(message)s", datefmt="%a, %d %b %Y %H:%M:%S"
     17 )
     18 LOG_ERROR_FORMATTER = logging.Formatter(
     19     fmt="[%(asctime)s] [%(levelname)-8s] [%(name)s]
[%(filename)s] [%(lineno)s] %(message)s",
     20     datefmt="%a, %d %b %Y %H:%M:%S",
     21 )

File /lib/python311.zip/logging/__init__.py:2050, in basicConfig(**kwargs)
   2048     else:
   2049         encoding = io.text_encoding(encoding)
-> 2050     h = FileHandler(filename, mode,
   2051                     encoding=encoding, errors=errors)
   2052 else:
   2053     stream = kwargs.pop("stream", None)

File /lib/python311.zip/logging/__init__.py:1181, in
FileHandler.__init__(self, filename, mode, encoding, delay, errors)
   1179     self.stream = None
   1180 else:
-> 1181     StreamHandler.__init__(self, self._open())

File /lib/python311.zip/logging/__init__.py:1213, in FileHandler._open(self)
   1208 """
   1209 Open the current base file with the (original) mode and encoding.
   1210 Return the resulting stream.
   1211 """
   1212 open_func = self._builtin_open
-> 1213 return open_func(self.baseFilename, self.mode,
   1214                  encoding=self.encoding, errors=self.errors)

OSError: [Errno 28] Invalid argument: '/drive/drive-2023-05-26-08-46-16.log'
```

I removed some of the logging by not initialising logging from `pgfinder/__init__.py` and instead moving it to the Command Line Interface `pgfinder/find_pg.py`. This way when I run the notebook there is no output, the buttons just change.

**Comment by bobturneruk** [comment]
Turns out I hadn't **fully** cleared the cache. I'll do a quick PR on the notebook repo...

**Pull request merged**